lib/client/backend_client.dart

```diff
@@ -1,23 +1,51 @@
+//import 'dart:convert';
+
+import 'dart:convert';
+
 import 'package:http/http.dart' as http;
 
 class BackendClient {
   const BackendClient();
 
 //functions to connect with API
-
-  Future<dynamic> getData({required String uri}) async {
+  Future<List<Map<String, String>>> getData({required String uri}) async {
     var url = Uri.http('10.0.2.2:8080', uri);
     try {
       var response = await http.get(url);
 
       if (response.statusCode == 200) {
-        return response.body;
+        dynamic decodedData = jsonDecode(response.body);
+
+        // Print the decoded data for inspection
+        print('Decoded data: $decodedData');
+
+        // Check if the decoded data is a List<Map<String, dynamic>>
+        if (decodedData is List &&
+            decodedData.isNotEmpty &&
+            decodedData[0] is Map) {
```

Comment:
Given my comment above, I would change this to just `decodedData.isNotEmpty`

Reply:
is this still needed with my most recent change?